### PR TITLE
Code Health - Add documentation for SNI APIs

### PR DESCRIPTION
**Pull request by jon-oracle**
##### Checklist
- [x] documentation is added or updated

##### Description of change
Add documentation for SNI APIs. Also tweaked find-doc-nits to display usage when called with no args.

### CODE REVIEW

File: util/find-doc-nits [hunk 1]
```diff
@@ -344,8 +344,7 @@ getopts('lnshu');
 
 &help() if ( $opt_h );
 
-die "Need one of -l -n -s or -u flags.\n"
-    unless $opt_l or $opt_n or $opt_s or $opt_u;
+&help() unless $opt_l or $opt_n or $opt_s or $opt_u;
```
Comment (levitte) [inline util/find-doc-nits]: Please remove this change.  If you really want it, make it a separate PR.  Personally, I find the change unhelpful.

Comment (jon-oracle) [inline util/find-doc-nits]: Yeah I'll remove it for now

**Comment by levitte** [comment]
Merged.

master: 8c55c4614731487449a02f90ff9a209ff6c63447
1.1.0: c6f2dc757f2456c788cf62323f0a434936896710
1.0.2: 144ab9b27ffec3f4f01a4a2c069309323bf886d9

**Comment by levitte** [comment]
Merged.

master: 8c55c4614731487449a02f90ff9a209ff6c63447
1.1.0: c6f2dc757f2456c788cf62323f0a434936896710
1.0.2: 144ab9b27ffec3f4f01a4a2c069309323bf886d9
